### ISSUE TITLE
Cannot authenticate to bitbucket cloud

### ISSUE THREAD

**Issue by chrdunne** [xebialabs-community/xlr-bitbucket-plugin#4]
Hi

I just installed the latest release of the plugin, but when I try to connect to Bitbucket I get a 403 (Permission) error.

I am using the URL https://api.bitcloud.org, and I am using the username and password I normally use to access the Bitbucket cloud UI.

I have 2FA enabled in bitbucket, but I don't think this should prevent access via the API.

Can you help me?

Thanks
Chris

**Comment by jdewinne** [comment]
When I try opening that URL, I'm not able to resolve the dns. Is that an internal url only?

**Comment by chrdunne** [comment]
Apologies - I meant that I am having the problem with this https://api.bitbucket.org. I also tried https://bitbucket.org and https://bitbucket.org/chris_dunne.

**Comment by jdewinne** [comment]
ok. In that case you'll need to use an [App password](https://blog.bitbucket.org/2016/06/06/app-passwords-bitbucket-cloud/). And when using an `App password` the username won't be your email, but your actual username.

![screenshot from 2018-01-05 08-21-57](https://user-images.githubusercontent.com/1579188/34617888-b26826e2-f1f1-11e7-819b-14cacc3d95f7.png)

**Comment by chrdunne** [comment]
Thanks for the quick response.

I have tried that too. My username is chris_dunne according to the settings in Bitbucket, but I still get a 403.

So finally I tried disabling two factor authentication, and this time it worked.

But I need two-factor auth. After further research  I found that Bitbucket has App Passwords. I generated an App Password and use that with my username. And this time the XL Release plugin successfully authenticated.
